pkg/apis/ceph.rook.io/v1/object.go

```diff
@@ -26,12 +26,14 @@ import (
 // will be registered for the validating webhook.
 var _ webhook.Validator = &CephObjectStore{}
 
+const serviceServingCertKey = "service.beta.openshift.io/serving-cert-secret-name"
+
 func (s *ObjectStoreSpec) IsMultisite() bool {
 	return s.Zone.Name != ""
 }
 
 func (s *ObjectStoreSpec) IsTLSEnabled() bool {
-	return s.Gateway.SecurePort != 0 && s.Gateway.SSLCertificateRef != ""
+	return s.Gateway.SecurePort != 0 && (s.Gateway.SSLCertificateRef != "" || len(s.GetServiceServingCert()) != 0)
```

Comment:
This returns a string, so checking for empty string (the same as with SSLCertificateRef) is the same as checking len==0 (but shorter).

```suggestion
	return s.Gateway.SecurePort != 0 && (s.Gateway.SSLCertificateRef != "" || s.GetServiceServingCert() != "")
```

Reply:
Add unit test for this method.